library/common-library/templates/_security-context.tpl

```diff
@@ -0,0 +1,40 @@
+{{- /* Defines the container securityContext context */ -}}
+{{- define "common.securityContext.container" -}}
+{{- $localSecurityContext := dict -}}
+{{- if get .Values "containerSecurityContext" | kindIs "map" -}}
+    {{- $localSecurityContext = .Values.containerSecurityContext -}}
+{{- end -}}
+
+{{- $globalSecurityContext := dict -}}
+{{- $global := index .Values "global" | default dict -}}
+{{- if get $global "containerSecurityContext" | kindIs "map" -}}
+    {{- $globalSecurityContext = $global.containerSecurityContext -}}
+{{- end -}}
+
+{{- if $localSecurityContext -}}
+    {{- toYaml $localSecurityContext -}}
+{{- else if $globalSecurityContext -}}
+    {{- toYaml $globalSecurityContext -}}
+{{- end -}}
+{{- end -}}
+
+
+{{- /* Defines the pod securityContext context */ -}}
+{{- define "common.securityContext.pod" -}}
+{{- $localSecurityContext := dict -}}
+{{- if get .Values "podSecurityContext" | kindIs "map" -}}
+    {{- $localSecurityContext = .Values.podSecurityContext -}}
+{{- end -}}
+
+{{- $globalSecurityContext := dict -}}
+{{- $global := index .Values "global" | default dict -}}
+{{- if get $global "podSecurityContext" | kindIs "map" -}}
+    {{- $globalSecurityContext = $global.podSecurityContext -}}
+{{- end -}}
```

Comment:
to me is a bit verbose just to avoid a nesting level below (and perhaps because I'm not used to program in tpl), but i'm fine since we have the test now 😃

Reply:
It tests that exists, is a map, and does not fail in case of being a null (which you reminded me that there is no test for that.

So I prefer that verbosity than having to see if `sprig` does a `default` left to right or right to left...

Reply:
In this case if not a map it will no fail during installation but the parameters are not going to be added (silently), but in case we just copy the yaml under the parameter and is not a map it will fail during installation. Am I right?